[WIP] Add new docs with Sphinx (.md)

chuneuny-emily opened the pull request:
<!-- Contributing guide: https://github.com/openvinotoolkit/datumaro/blob/develop/CONTRIBUTING.md -->

### Summary
This PR contain the restructured documentation with Sphinx.

<!--
Resolves #111 and #222.
Depends on #1000 (for series of dependent commits).

This PR introduces this capability to make the project better in this and that.

- Added this feature
- Removed that feature
- Fixed the problem #1234
-->

### How to test
<!-- Describe the testing procedure for reviewers, if changes are
not fully covered by unit tests or manual testing can be complicated. -->

### Checklist
<!-- Put an 'x' in all the boxes that apply -->
- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

chuneuny-emily committed:
Add new docs with sphinix

chuneuny-emily committed:
Fix lint isse

chuneuny-emily committed:
Remove log

chuneuny-emily committed:
Correct CVAT github repo

wonjuleee commented on docs/emily.log:
```diff
@@ -0,0 +1,552 @@
+Running Sphinx v5.3.0
```
Should this be removed?

chuneuny-emily replied:
> Should this be removed?
Ops. I removed it now.

vinnamkim commented on docs/emily.log:
Please remove this file.

chuneuny-emily replied:
done

vinnamkim commented on .github/workflows/github_pages.yml:
```diff
@@ -3,7 +3,7 @@ name: Github pages
   push:
     branches:
-      - releases/v1.0.0
+      - releases/**
```
Why don't we build docs for `develop` change also?

chuneuny-emily replied:
I applied the change from https://github.com/openvinotoolkit/datumaro/pull/881. @yunchu Can you check it?

yunchu replied:
In case of develop, need to add separate workflow for the new docs because the theme to be used for the new docs would not have the feature like "docs per release version".

vinnamkim replied:
The develop only pipeline is planned to be added in the future, right?

yunchu replied:
yes, right. and that will generate docs and exposed as 'latest' through the different URL from the release one (stable).

chuneuny-emily committed:
Correct CVAT github repo

wonjuleee commented on docs/source/docs/get-started/quick-start-guide/examples.rst:
```diff
@@ -0,0 +1,98 @@
+                  -- --reindex 1 # avoid annotation id conflicts
+
+- Convert only non-``occluded`` annotations from a
+  `CVAT <https://github.com/openvinotoolkit/cvat>`_ project to TFrecord:
```
should this be updated? https://github.com/opencv/cvat

chuneuny-emily replied:
> should this be updated? https://github.com/opencv/cvat

I corrected the URL now.

wonjuleee commented on docs/source/docs/get-started/quick-start-guide/examples.rst:
```diff
@@ -0,0 +1,98 @@
+                  -- --reindex 1 # avoid annotation id conflicts
+
+- Convert only non-``occluded`` annotations from a
+  `CVAT <https://github.com/openvinotoolkit/cvat>`_ project to TFrecord:
```
This might be redundant for now.

chuneuny-emily replied:
> This might be redundant for now.

I corrected all old cvat repo URL in docs.

wonjuleee commented on docs/source/docs/get-started/quick-start-guide/examples.rst:
```diff
@@ -0,0 +1,98 @@
+    datum export --format tf_detection_api -- --save-images
+
+- Annotate MS COCO dataset, extract image subset, re-annotate it in
+  `CVAT <https://github.com/openvinotoolkit/cvat>`_, update old dataset:
```
we may be possible to remove re-annotate part in CVAT

chuneuny-emily replied:
Is there a reason you want to delete the re-annotate part? It explains how to update the new annotation in Datumaro, and I think it's good to have one. e.g., datum project update reannotation-upd

chuneuny-emily committed:
Correct CVAT github repo

wonjuleee reviewed (commented):
We agreed to merge the doc-related PRs to `docs-rework` branch first, don't we?

vinnamkim commented on docs/source/docs/get-started/quick-start-guide/installation.rst:
```diff
@@ -0,0 +1,83 @@
+- PyTorch
+- MxNet
+- Caffe
+- Usag
```
It seems a typo.

chuneuny-emily replied:
done.

vinnamkim commented on docs/source/docs/reference/jupyter_notebook_examples/generate.rst:
```diff
@@ -1,8 +1,5 @@
-linkTitle: 'generate'
-description: ''
----
+generate
```
```suggestion
Generate
```

chuneuny-emily replied:
> 

done

vinnamkim commented on docs/source/docs/reference/jupyter_notebook_examples/filter.rst:
```diff
@@ -0,0 +1,53 @@
+filter
```
```suggestion
Filter
```

chuneuny-emily replied:
> 

done

vinnamkim commented on docs/source/docs/reference/jupyter_notebook_examples/explore.rst:
```diff
@@ -0,0 +1,8 @@
+explore
```
```suggestion
Explore
```

vinnamkim commented on docs/source/docs/reference/jupyter_notebook_examples/dataset_handler.rst:
```diff
@@ -1,8 +1,6 @@
-description: ''
----
+
+dataset-handler
```
```suggestion
Dataset-handler
```

vinnamkim commented on docs/source/docs/reference/jupyter_notebook_examples/tiling.rst:
```diff
@@ -1,8 +1,5 @@
-linkTitle: 'tiling'
-description: ''
----
+tiling
```
```suggestion
Tiling
```

vinnamkim commented on docs/source/docs/reference/jupyter_notebook_examples/index.rst:
```diff
@@ -0,0 +1,17 @@
+.. toctree::
+   :maxdepth: 1
+
+   dataset_handler
+   filter
+   tiling
+   transform
+   visualizer
+   explore
+   e2e_example
```
Do we have to manually list all the items by hands? Is it possible such as `notebooks/*`?

done

vinnamkim commented on docs/source/docs/reference/jupyter_notebook_examples/transform.rst:
```diff
@@ -0,0 +1,8 @@
+transform
```
```suggestion
Transform
```

vinnamkim commented on docs/source/docs/reference/jupyter_notebook_examples/validate.rst:
```diff
@@ -1,8 +1,5 @@
-linkTitle: 'validate'
-description: ''
----
+validate
```
```suggestion
Validate
```

vinnamkim commented on docs/source/docs/reference/jupyter_notebook_examples/visualizer.rst:
```diff
@@ -1,8 +1,5 @@
-linkTitle: 'visualizer'
-description: ''
----
+visualizer
```
```suggestion
Visualizer
```

vinnamkim commented on docs/source/docs/user-manual/command-reference/checkout.md:
```diff
@@ -1,8 +1,4 @@
-linkTitle: 'checkout'
-description: ''
----
+# checkout
```
Please capitalize the title. There are also so many below too.

chuneuny-emily replied:
It shows the original command. If I change it to capital, then menus will show as capital.

chuneuny-emily committed:
Apply feedbacks

chuneuny-emily commented:
> docs-rework

Yes, but the work in the form of the previous copy was already in progress there, so I got confused and took a branch again from the latest develop. You can also delete the existing docs-rework branch and rename it to that one.

chuneuny-emily commented:
> We agreed to merge the doc-related PRs to `docs-rework` branch first, don't we?

Yes, but the work in the form of the previous copy was already in progress there, so I got confused and took a branch again from the latest develop. You can also delete the existing docs-rework branch and rename it to that one.

chuneuny-emily committed:
Add docs for new features

chuneuny-emily committed:
Fix whitespace issue

chuneuny-emily committed:
Add github icon link in head

chuneuny-emily committed:
Fix Meramaid graph

chuneuny-emily committed:
Merge from develop

chuneuny-emily committed:
Change tab to space in Makefile

chuneuny-emily committed:
Revert index

yunchu committed:
Update .gitattibutes file (#893)

To exclude 'tab-in-indent' error from whitespace checking on Makefile

chuneuny-emily committed:
Fix broken links

chuneuny-emily committed:
Merge branch 'develop' of https://github.com/openvinotoolkit/datumaro into sphinx-docs-rework

chuneuny-emily committed:
Merge branch 'sphinx-docs-rework' of https://github.com/openvinotoolkit/datumaro into sphinx-docs-rework

chuneuny-emily committed:
Fix pre-commit issue

chuneuny-emily committed:
Update API doc

chuneuny-emily committed:
Rebase develop

wonjuleee committed:
[Doc] Reorganize content structure (#908)

<!-- Contributing guide:
https://github.com/openvinotoolkit/datumaro/blob/develop/CONTRIBUTING.md
-->

### Summary


![image](https://user-images.githubusercontent.com/89109581/229419480-c4835066-9253-4a48-8a98-6630e951c34b.png)

<!--
Resolves #111 and #222.
Depends on #1000 (for series of dependent commits).

This PR introduces this capability to make the project better in this
and that.

- Added this feature
- Removed that feature
- Fixed the problem #1234
-->

### How to test
<!-- Describe the testing procedure for reviewers, if changes are
not fully covered by unit tests or manual testing can be complicated.
-->

### Checklist
<!-- Put an 'x' in all the boxes that apply -->
- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into
[CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the
[documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs)
accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT
License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE)
that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example
below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

wonjuleee committed:
[Doc] reformat contents and add level up skills (#916)

<!-- Contributing guide:
https://github.com/openvinotoolkit/datumaro/blob/develop/CONTRIBUTING.md
-->

### Summary

<!--
Resolves #111 and #222.
Depends on #1000 (for series of dependent commits).

This PR introduces this capability to make the project better in this
and that.

- Added this feature
- Removed that feature
- Fixed the problem #1234
-->

### How to test
<!-- Describe the testing procedure for reviewers, if changes are
not fully covered by unit tests or manual testing can be complicated.
-->

### Checklist
<!-- Put an 'x' in all the boxes that apply -->
- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into
[CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the
[documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs)
accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT
License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE)
that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example
below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

yunchu commented:
Changes reviewed on the PR was applied #917 and merged.
